### PR TITLE
Configurable UMFPACK iterative refinements #72

### LINKED ISSUE #72: Support for changing number of UMFPACK iterative refinements
The number of iterative refinements in UMFPACK defaults to 2. By lowering this value, it is possible to greatly speed up the solver. For many applications, one iterative refinement might be enough (for the example below, the second refinement is not needed), while zero iterative refinements might compromise too much accuracy. Changing this value could speed up methods that rely heavily on system solves (such as NLEIGS). We could perhaps make this configurable in `DefaultLinSolver`.

```julia
julia> using SuiteSparse, LinearAlgebra, BenchmarkTools

julia> n = 10000;

julia> A = I + sprandn(n, n, 0.0005);

julia> y = randn(n);

julia> luf = lu(A);

julia> @btime global x = luf \ y; norm(A*x - y)
  52.467 ms (7 allocations: 625.20 KiB)
1.2050980166311244e-12

julia> SuiteSparse.UMFPACK.umf_ctrl[8] = 1;

julia> @btime global x = luf \ y; norm(A*x - y)
  34.722 ms (7 allocations: 625.20 KiB)
1.2050980166311244e-12

julia> SuiteSparse.UMFPACK.umf_ctrl[8] = 0;

julia> @btime global x = luf \ y; norm(A*x - y)
  15.467 ms (7 allocations: 312.70 KiB)
4.9111032477641746e-9
```

### PR DESCRIPTION
**Pull request by maxbennedich**
Fixes #72 

I removed the matrix `A` from `DefaultLinSolver` since it wasn't used for anything, and also removed some parametric types since as far as I could tell they didn't add anything; please let me know if I'm missing something!

### CODE REVIEW

File: src/LinSolvers.jl [hunk 1]
```diff
@@ -40,29 +41,42 @@ module LinSolvers
...
     end
-    DefaultLinSolver(A::AbstractMatrix,Afact) = new(A, Afact)
 
-    function DefaultLinSolver(nep::NEP,λ)
+    function DefaultLinSolver(nep::NEP, λ, umfpack_refinements)
         A=compute_Mder(nep,λ)
         Afact=factorize(A)
-        return DefaultLinSolver{eltype(A),typeof(A)}(A, Afact)
+        return DefaultLinSolver(Afact, umfpack_refinements)
     end
```
Comment (jarlebring) [inline src/LinSolvers.jl]: I also don't directly see the use of having parametric types here. It was turned parametric at a point before we fully understood how to use parametric types well (march) 

However, in #44 we made newton + DEP type stable. With these type of changes might loose that? To be sure not to loose this along the way, we could add a unit test corresponding verifying type stability similar to #44.

Comment (jarlebring) [reply src/LinSolvers.jl]: I meant `augnewton` is type stable for DEPs: 
```
julia> dep=nep_gallery("dep0");
julia> @code_warntype(augnewton(Float64,dep))
```
So a short unit test checking could something like:
```
dep=nep_gallery("dep0");
Test.@inferred augnewton(Float64,dep,v=ones(size(dep,1)))
Test.@inferred augnewton(ComplexF64,dep,v=ones(size(dep,1)))

```

Comment (eringh) [reply src/LinSolvers.jl]: Yes, we did not fully understand how to use parametric types. However, considering the following [performance tips](https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-fields-with-abstract-type-1) in the Julia manual it is possible that we still want to have the types parametric. Although maybe something like
```
    struct DefaultLinSolver{T} <: LinSolver
        Afact::T
        umfpack_refinements::Int
    end

    function DefaultLinSolver(nep::NEP, λ, umfpack_refinements))
        A = compute_Mder(nep,λ)
        Afact = factorize(A)
        return DefaultLinSolver{typeof(Afact)}(Afact, umfpack_refinements))
    end
```

Comment (maxbennedich) [reply src/LinSolvers.jl]: Good suggestions. I've updated the code accordingly.

**Pull request merged**